src/main/java/core/basesyntax/bookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.bookstore.repository.impl;
+
+import core.basesyntax.bookstore.model.Book;
+import core.basesyntax.bookstore.repository.BookRepository;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();

```

Comment:
You have Spring transaction manager already, use `@Transactional` instead of manual transactions handling